src/query.ts

```diff
@@ -597,10 +599,17 @@ export interface IntegerTypeCastOptions {
   properties?: string | string[];
 }
 
+export enum QueryMode {
```

Comment:
I think this is old design. In the new design, the user will pass ExplainOptions. Please refer 'Query Profiling for Datastore SDKs' document created by @kolea2

Reply:
Great catch. I made updates to reflect this change.